Sofa/Component/Mass/src/sofa/component/mass/UniformMass.inl

```diff
@@ -82,31 +82,35 @@ UniformMass<DataTypes>::UniformMass()
 
     sofa::core::objectmodel::Base::addUpdateCallback("updateFromTotalMass", {&d_totalMass}, [this](const core::DataTracker& )
     {
-        if(m_isTotalMassUsed)
+        if(m_initMethod == InitMethod::TOTALMASS)
         {
             msg_info() << "dataInternalUpdate: data totalMass has changed";
             return updateFromTotalMass();
         }
-        else
+        else if(m_initMethod == InitMethod::VERTEXMASS)
         {
             msg_info() << "vertexMass data is initially used, the callback associated with the totalMass is skipped";
             return updateFromVertexMass();
         }
+        else
+            return sofa::core::objectmodel::ComponentState::Invalid;
     }, {});
 
 
     sofa::core::objectmodel::Base::addUpdateCallback("updateFromVertexMass", {&d_vertexMass}, [this](const core::DataTracker& )
     {
-        if(!m_isTotalMassUsed)
+        if(m_initMethod == InitMethod::VERTEXMASS)
         {
             msg_info() << "dataInternalUpdate: data vertexMass has changed";
             return updateFromVertexMass();
         }
-        else
+        else if(m_initMethod == InitMethod::TOTALMASS)
         {
             msg_info() << "totalMass data is initially used, the callback associated with the vertexMass is skipped";
             return updateFromTotalMass();
         }
+        else
```

Comment:
This looks kind of a duplication.

Reply:
Here is a version with less duplication: https://github.com/sofa-framework/sofa/pull/5429